controllers/api_v1/keyword.go

```diff
@@ -18,9 +19,29 @@ const (
 type KeywordAPIController struct{}
 
 func (kapi *KeywordAPIController) ApplyRoutes(engine *gin.RouterGroup) {
+	engine.GET("/keywords", kapi.fetchKeywords)
 	engine.POST("/keywords", kapi.uploadKeyword)
 }
 
+func (kapi *KeywordAPIController) fetchKeywords(c *gin.Context) {
+	currentUserID := helpers.GetCurrentUserID(c)
+	keywordService := keyword_service.KeywordService{CurrentUserID: currentUserID}
+	keywords, err := keywordService.GetAll()
+
+	if err != nil {
+		errorResponse := api_helper.ErrorResponseObject{
+			Detail: err.Error(),
+			Status: http.StatusBadRequest,
+		}
+		c.JSON(errorResponse.Status, errorResponse.NewErrorResponse())
+		return
+	}
+
+	keywordsResponse := keyword_api_service.KeywordsResponse{Keywords: keywords}
```

Comment:
`keyword_api_service` does not do anything outside formatting the JSON response, there are not even DB work. So it's not a real service.

That's where serializers are needed. They are presenters for APIs. If the rendering is simple, it can be done in the controllers. For complex rendering, use serializers.

You can see some examples in The 1 API for instance: https://github.com/nimblehq/the1-api/tree/development/engines/the1_idm/app/serializers/the1_idm/v2. Here is [an intro article](https://dev.to/amnyrrd/json-serializers-in-a-rails-api-5262) about its usage with Rails.

Reply:
Added `serializers` in f886c13c263f65571a2d083228fb95009ec62634